### DISCUSSION TITLE
When use Liquidsoap with Webcaster(webcast.js) the audio has a bad quality

### DISCUSSION THREAD

**Discussion by caf2022** [savonet/liquidsoap#2505] [General]
When use Liquidsoap with Webcaster(webcast.js) the audio has a bad quality, is this a problem with webcast.js or liquidsoap configuration?

I'm tested with the https://webcast.github.io/webcaster/ demo and with downloaded files in my server too.

**Comment by toots** [comment]
Hi!

With the latest webcast online implementation, audio is encoded using the browser's web audio framework so this is where I would look first. If such a thing exist, it would be nice to compare to another webapp making use of the web audio framework to e.g. create audio files.

**Reply by caf2022** [reply]
I see that azuracast uses the same but now have this problem and have the mp3 encoder option, Is possible to get the old version of webcast ? On the github I can't find. Thanks for the help.

**Reply by toots** [reply]
On my browser I only have a `opus` option. What browser are you using?

**Comment by caf2022** [comment]
Firefox, chrome and edge, to me only show the opus too, but on azuracast show mp3 and I see that on old version had the mp3 option, maybe using a JS framework/lib

Is possible to get this JS files from the older version?

https://github.com/webcast/webcast.github.io/blob/master/client/index.html

**Reply by toots** [reply]
Yes, you can definitely checkout an old commit and use that code.

**Reply by caf2022** [reply]
Is posible to get this files? How?

    <script src="https://rawgithub.com/webcast/webcast.js/master/lib/webcast.js"></script>
    <script src="https://rawgithub.com/webcast/webcast.js/master/examples/client/client.js"></script>
